src/vs/base/common/glob.ts

```diff
@@ -264,6 +268,16 @@ export function match(arg1: string | IExpression, path: string, siblings?: strin
 
 	// Glob with String
 	if (typeof arg1 === 'string') {
+
+		if (trivia1.test(arg1)) {
+			// common pattern: **/*.txt just need endsWith check
+			return strings.endsWith(path, arg1.substr(4)); // '**/*'.length === 4
```

Comment:
you are assuming that most of these patterns have 3 letters in file type, but we know that it is not always the case, like `.js`, `.ts`. Better will be to actually get the match

```
const trivia1 = /^\*\*\/\*(\.\w+)$/;

if ((let t1match = trivia1.match(arg1)) != null) {
    return strings.endsWith(path, t2match[1]);
}
```